docs/testing/e2e-testing-standards.md

```diff
@@ -0,0 +1,259 @@
+
+        - `config/`: Configuration files for Playwright and other tools.
+
+            - `playwright.config.ts`: Playwright configuration.
+
+        - `hooks/`: Cucumber hooks for setting up and tearing down test environments.
+
+            - `hooks.ts`: Contains `Before`, `After`, and other lifecycle hooks.
+
+        - `reports/`: Generated test reports (e.g., HTML, JSON).
+
+            - `screenshots/`: Captured screenshots during test execution.
+
+            - `videos/`: Recorded videos of test runs.
+
+        - `test-data/`: Static test data files or folders for upload.
+
+            - `data.json`: Example JSON files or CSVs with test data.
+
+## Test Structure - Arrange, Act, Assert:
+
+We can follow the AAA (Arrange, Act, Assert) pattern when structuring the tests. In most cases, the Arrange step can be included in a Before block(hook).
+Consider including comments defining each section to ease readability.
+
+```typescript
+// arrange, set up the initial conditions for the test
+// create a property
+await createProperty()
+
+// act, perform the action that you want to test 
+// raise a charge
+// this could involve calling methods or functions defined in your page objects
+await raiseCharge()
+
+// assert, verify that the action had the expected outcome 
+// confirm charge has been raised
+expect(charge).toBe('raised')
+```
+
+## Page Object Model (POM)
+
+Every page should possess one POM file to enhance maintainability and scalability of our tests. It should include all the selectors and functions which are needed to interact with the UI page or component(s).
+
+All interactions should be done using the page objects, no selectors in your tests.
+
+All assertions should be in your test, no assertion in the POM
+
+DO 👍
+
+```typescript
+// POM file './pageOobjects/foo/'
+// add all locators and functions related to the page. 
+// allowing all tests to reuse 
+
+import {expect, Locator, Page} from '@playwright/test';
+
+export class FooPage {
+    readonly errorMessage: Locator;
+
+    constructor(page: Page) {
+        this.page = page;
+        this.errorMessage = page.locator('.error-message');
+    }
+}
+
+// test file './steps/foo.ts'
+import {FooPage} from './pageObjects/foo';
+
+let fooPage: FooPage;
+
+Then(
+    'error message should be visible',
+    async function ({page}) {
+        const fooPage = new FooPage({page});
+        await expect(fooPage.errorMessage).toBeVisible();
+    }
+)
+```
+
+DO NOT ⚔️
+```typescript
+// test file './steps/foo.ts'
+// include locators directly in test
+import { Locator, Page } from '@playwright/test';
+
+Then(
+    'error message should be visible',
+    async function ({page}) {
+        await expect(page.locator('.error-message')).toBeVisible();
+    }
+)
+```
+
+## Waiting:
+
+We can avoid using any arbitrary waits until it is really, really necessary.
+
+DO NOT ⚔️
+
+```typescript
+await page.waitForTimeout(5000)
+```
```

Comment:
we don't need set artificial waiting like `waitForTimeout(5000)` Playwright uses Auto-waiting.

t's a sentence from from playwright docs:
 Playwright waits for elements to be actionable prior to performing actions. It also has a rich set of introspection events. The combination of the two eliminates the need for artificial timeouts - the primary cause of flaky tests.

Reply:
Note: this is a "DO NOT DO THIS" example - the line above has "DO NOT"

Reply:
yes, I suggested to change line `We can avoid using any arbitrary waits until it is really, really necessary.` as `Playwright uses auto-waiting, so we avoid artificial waiting, the exception being if it is really necessary.`